### PR TITLE
Correct error message when adding/removing processes from workers

### LINKED ISSUE #20742: Adding Julia workers from other workers
Ideally this would work, but it should at least report a legible error:

```julia
julia> addprocs(1)
1-element Array{Int64,1}:
 2

julia> remotecall_wait(2) do
       addprocs(1)
       end
ERROR: RemoteException(2, CapturedException(CompositeException(Any[CapturedException(AssertionError(""), Any[(assert at error.jl:68 [inlined], 1), (create_worker at cluster.jl:401, 1), (setup_launched_worker at cluster.jl:353, 1), (#31 at task.jl:335, 1)])]), Any[(sync_end at task.jl:287, 1), (macro expansion at task.jl:303 [inlined], 1), (#addprocs_locked#28 at cluster.jl:306, 1), (#addprocs#27 at cluster.jl:283, 1), (#addprocs#231 at managers.jl:312, 1), (#1 at REPL[2]:2, 1), (#100 at process_messages.jl:275, 1), (run_work_thunk at process_messages.jl:56, 1), (run_work_thunk at process_messages.jl:65 [inlined], 1), (#86 at event.jl:73, 1)]))
Stacktrace:
 [1] #remotecall_wait#136(::Array{Any,1}, ::Function, ::Function, ::Base.Distributed.Worker) at ./distributed/remotecall.jl:382
 [2] remotecall_wait(::Function, ::Base.Distributed.Worker) at ./distributed/remotecall.jl:373
 [3] #remotecall_wait#139(::Array{Any,1}, ::Function, ::Function, ::Int64) at ./distributed/remotecall.jl:394
 [4] remotecall_wait(::Function, ::Int64) at ./distributed/remotecall.jl:394
```

### PR DESCRIPTION
**Pull request by amitmurthy**
Closes #20742

### CODE REVIEW

File: test/distributed_exec.jl [hunk 1]
```diff
@@ -1272,6 +1272,16 @@ if DoFullTest
     rmprocs(pids)
 end
 
+# Test addprocs/rmprocs from master node only
+for f in [ ()->addprocs(1), ()->rmprocs(workers()) ]
+    try
+        remotecall_fetch(f, id_other)
```
Comment (tkelman) [inline test/distributed_exec.jl]: this needs an `error("unexpected")` otherwise it'll silently do nothing if it succeeds

Comment (amitmurthy) [inline test/distributed_exec.jl]: yes. good catch.

Comment (tkelman) [inline test/distributed_exec.jl]: got it in #21059

Comment (amitmurthy) [inline test/distributed_exec.jl]: Thanks!

**Pull request merged**